algebra-kernels/src/msm.rs

```diff
@@ -0,0 +1,344 @@
+use algebra_cl_gen::gpu::{GPUError, GPUResult, get_core_count, get_prefix_map, kernel_multiexp};
```

Comment:
Maybe it's worth to have a couple of unit tests asserting equality between the GPU result and the CPU one ?

Reply:
Inside ginger-lib we have now "test_all_variants" and "test_with_unequal_numbers" tests to check the results

Reply:
I would've preferred two separate tests but it's fine